src/actions.js

```diff
@@ -96,21 +99,40 @@ export function fetchOrganisations() {
     } else {
       dispatch(selectOrganisation(selectedOrganisationLocalStorage));
     };
+
+    // request contracts
+    const requestContracts = (url) => {
+      dispatch({
+        type: REQUEST_CONTRACTS,
+      });
+      fetch(url, {
+          credentials: "same-origin"
+      })
+      .then(response => response.json())
+      .then(data => {
+          dispatch({
+            type: SET_CONTRACTS,
+            contracts: data,
+          });
+          if (data.next) {
+            const relativeUrl = data.next.split('lizard.net')[1];
+            requestContracts(relativeUrl);
+          }
+      });
+    };
+    requestContracts(`/api/v4/contracts`);
```

Comment:
I made a `recursiveFetchFunction` in the hooks.ts file. I think it can be used here too. An example of where it is used is the fetchOrganisations function in action.ts.

Reply:
done. Good point actually ;)